### PR TITLE
Lourdes/battleship

### PR DESCRIPTION
**Pull request by lourdesVillca**

### CODE REVIEW

**Comment by lourdesVillca** [comment]
image

File: src/test/java/org/minions/devfund/lourdes/battleship/OceanTest.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            assertTrue(ocean.isOccupied(row, i));
+        }
+    }
+
+    /**
+     * Verify if a place is not occupied.
+     */
+    @Test
+    public void testIsNotOccupied() {
+        Ocean ocean = new Ocean();
```
Comment (carledriss) [inline src/test/java/org/minions/devfund/lourdes/battleship/OceanTest.java]: MAJOR Rename "ocean" which hides the field declared at line 13. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)

File: src/main/java/org/minions/devfund/lourdes/battleship/Ship.java [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+     * @param ocean      Ocean object where the ship will be placed.
+     * @return True if the ship is able to be placed in the given position, false otherwise.
+     */
+    private boolean checkFreeSpaceForShip(int row, int column, boolean horizontal, final Ocean ocean) {
+        try {
+            for (int i = 0; i < length; i++) {
+                if (horizontal && ocean.isOccupied(row, column + i)) {
+                    return false;
+
+                } else if (ocean.isOccupied(row + i, column)) {
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/lourdes/battleship/Ship.java]: MAJOR This branch's code block is the same as the block for the branch on line 117. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1871)

File: src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+            }
+        }
+    }
+
+    /**
+     * Method that place randomly all ships in the ocean.
+     */
+    void placeAllShipsRandomly() {
+        ShipFactory shipFactory = new ShipFactory();
+        SHIP_TYPE.forEach((key, value) -> {
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java]: MINOR Remove useless curly braces around statement [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1602)

**Comment by carledriss** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    public void print() {
+        printColumnPosition();
+        for (int i = 0; i < ships.length; i++) {
+            System.out.println("");
+            System.out.print(i);
+            for (int j = 0; j < ships.length; j++) {
+                if (ships[i][j].toString().equals("S") && !ships[i][j].hit[ships[i][j].getHitIndex(i, j)]) {
+                    System.out.print(" . ");
+                } else {
+                    System.out.print(" " + ships[i][j] + " ");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+                }
+            }
+            System.out.println("");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+        }
+        System.out.println("");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+        StringBuilder arrayPosition = new StringBuilder(" ");
+        for (int i = 0; i < ships.length; i++) {
+            if (i < maxNumberAddCero) {
+                arrayPosition.append("0");
+            }
+            arrayPosition.append(i);
+            arrayPosition.append(" ");
+
+        }
+        System.out.print(arrayPosition);
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+        return false;
+    }
+
+    /**
+     * Method that print the ocean.
+     */
+    public void print() {
+        printColumnPosition();
+        for (int i = 0; i < ships.length; i++) {
+            System.out.println("");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+            System.out.print(i);
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

File: src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+            for (int j = 0; j < ships.length; j++) {
+                if (ships[i][j].toString().equals("S") && !ships[i][j].hit[ships[i][j].getHitIndex(i, j)]) {
+                    System.out.print(" . ");
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/lourdes/battleship/Ocean.java]: MINOR Replace this use of System.out or System.err by a logger. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

**Comment by carledriss** [comment]
SonarQube analysis reported 10 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 8 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by carledriss** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by carledriss — CHANGES_REQUESTED** [review]
Increase the code coverage

**Review by carledriss — APPROVED** [review]

**Pull request merged**